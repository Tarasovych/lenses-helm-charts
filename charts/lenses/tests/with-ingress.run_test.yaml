suite: enable ingress and check that host is escaped correclty
templates:
  - ingress.yaml
tests:
  - it: should work
    set:
      labels:
        foo: bar
      ingress:
        enabled: true
        host: example.com
    asserts:
      - equal:
          path: spec.rules[0].host
          value: example.com
  - it: should work with asterisk
    set:
      labels:
        foo: bar
      ingress:
        enabled: true
        host: "*.example.com"
    asserts:
      - equal:
          path: spec.rules[0].host
          value: "*.example.com"
  - it: should work with Capability networking.k8s.io/v1 and api group contains Ingress resource
    capabilities:
      apiVersions:
        - networking.k8s.io/v1/Ingress
    release:
      name: bar
    set:
      ingress:
        enabled: true
    asserts:
      - equal:
          path: apiVersion
          value: networking.k8s.io/v1
      - equal:
          path: spec.rules[0].http.paths[0].pathType
          value: Prefix
      - equal:
          path: spec.rules[0].http.paths[0].backend.service.name
          value: bar-lenses
      - equal:
          path: spec.rules[0].http.paths[0].backend.service.port.number
          value: 80
  - it: should work without special capabilities
    release:
      name: bar
    set:
      ingress:
        enabled: true
    asserts:
      - equal:
          path: apiVersion
          value: extensions/v1beta1
      - isEmpty:
          path: spec.rules[0].http.paths[0].pathType
      - equal:
          path: spec.rules[0].http.paths[0].backend.serviceName
          value: bar-lenses
      - equal:
          path: spec.rules[0].http.paths[0].backend.servicePort
          value: 80
  - it: should work with Capability networking.k8s.io/v1beta1
    capabilities:
      apiVersions:
        - networking.k8s.io/v1beta1
    release:
      name: bar
    set:
      ingress:
        enabled: true
    asserts:
      - equal:
          path: apiVersion
          value: networking.k8s.io/v1beta1
      - isEmpty:
          path: spec.rules[0].http.paths[0].pathType
      - equal:
          path: spec.rules[0].http.paths[0].backend.serviceName
          value: bar-lenses
      - equal:
          path: spec.rules[0].http.paths[0].backend.servicePort
          value: 80
  - it: should work with Capability networking.k8s.io/v1 without Ingress, for clusters on 1.18 version
    capabilities:
      apiVersions:
        - networking.k8s.io/v1
        - networking.k8s.io/v1beta1
        - networking.k8s.io/v1beta1/Ingress
    set:
      ingress:
        enabled: true
    asserts:
      - equal:
          path: apiVersion
          value: networking.k8s.io/v1beta1
