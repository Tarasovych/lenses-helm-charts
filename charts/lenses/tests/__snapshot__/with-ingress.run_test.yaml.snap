should work:
  1: |
    apiVersion: extensions/v1beta1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: traefik
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.7
        component: lenses
        foo: bar
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-ingress
        release: RELEASE-NAME
      name: RELEASE-NAME
    spec:
      rules:
      - host: example.com
        http:
          paths:
          - backend:
              serviceName: RELEASE-NAME
              servicePort: 80
            path: /
should work with asterisk:
  1: |
    apiVersion: extensions/v1beta1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: traefik
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.7
        component: lenses
        foo: bar
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-ingress
        release: RELEASE-NAME
      name: RELEASE-NAME
    spec:
      rules:
      - host: '*.example.com'
        http:
          paths:
          - backend:
              serviceName: RELEASE-NAME
              servicePort: 80
            path: /
